backend/cn/main.ml

```diff
@@ -421,90 +438,138 @@ module Executable_spec_flags = struct
+  let doc = "[Experimental] Set the maximum number of unfolds for recursive predicates" in
+  Arg.(value & opt int 5 & info ["max-unfolds"] ~doc)
 
 let testing_framework =
-  let doc = "[Experimental] Specify the testing framework for test suite generation " in
-  Arg.(value & opt (enum ["gtest", TestGeneration.GTest]) GTest & info ["testing-framework"] ~doc)
+  let doc = "[Experimental] Specify the testing framework used by generated tests" in
+  Arg.(value & opt (enum ["gtest", TestGeneration.GTest]) GTest & info ["framework"] ~doc)
 
 end
 
-let () =
+let generate_tests_cmd =
   let open Term in
-  let check_t =
-    const main $
-      Common_flags.file $
-      Common_flags.macros $
-      Common_flags.incl_dirs $
-      Common_flags.incl_files $
-      Verify_flags.loc_pp $
-      Common_flags.debug_level $
-      Common_flags.print_level $
-      Common_flags.print_sym_nums $
-      Verify_flags.slow_smt_threshold $
-      Verify_flags.slow_smt_dir $
-      Common_flags.no_timestamps $
-      Verify_flags.json $
-      Verify_flags.state_file $
-      Verify_flags.output_dir $
-      Verify_flags.diag $
-      Lemma.lemmata $
-      Verify_flags.only $
-      Verify_flags.skip $
-      Verify_flags.csv_times $
-      Verify_flags.log_times $
-      Verify_flags.random_seed $
-      Verify_flags.solver_logging $
-      Verify_flags.solver_flags $
-      Verify_flags.solver_path $
-      Verify_flags.solver_type $
-      Executable_spec_flags.output_decorated_dir $
-      Executable_spec_flags.output_decorated $
-      Executable_spec_flags.with_ownership_checking $
-      Executable_spec_flags.copy_source_dir $
-      Common_flags.astprints $
-      Verify_flags.use_vip $
-      Common_flags.no_use_ity $
-      Verify_flags.use_peval $
-      Verify_flags.batch $
-      Common_flags.no_inherit_loc $
-      Common_flags.magic_comment_char_dollar $
-      Test_Generation_Flags.output_test_dir $
-      Test_Generation_Flags.test_max_unfolds $
-      Test_Generation_Flags.testing_framework
+  let generate_tests_t =
+    const generate_tests $
+    Common_flags.file $
+    Common_flags.macros $
+    Common_flags.incl_dirs $
+    Common_flags.incl_files $
+    Common_flags.astprints $
+    Verify_flags.use_peval $
+    Common_flags.no_inherit_loc $
+    Common_flags.magic_comment_char_dollar $
+    Test_generation_flags.output_test_dir $
+    Test_generation_flags.test_max_unfolds $
+    Test_generation_flags.testing_framework
   in
-  let version_str = "CN version: " ^ Cn_version.git_version in
-  let cn_info = Cmd.info "cn" ~version:version_str in
-  Stdlib.exit @@ Cmd.(eval (v cn_info check_t))
+  let doc = String.trim
+    "
+    Generates RapidCheck tests for all functions in [FILE] with CN specifications.
+    The tests use randomized inputs, which are guaranteed to satisfy the CN precondition.
+    A [.cpp] file containing the test harnesses will be placed in [output-dir].
+    "
```

Comment:
This can use quoted string literals or escaped newlines instead of trimming https://ocaml.org/manual/5.2/lex.html#sss:stringliterals